config/main.py

```diff
@@ -570,6 +570,82 @@ def is_ipaddress(val):
         return False
     return True
 
+#
+# Check if an interface_name is in a vlan
+#
+def  interface_is_in_vlan(vlan_member_table, interface_name):
+
+    for k,v in vlan_member_table:
+        if v == interface_name:
+            return True
+
+    return False
+
+
+#
+# Check if port is already configured as mirror destination port
+#
+def interface_is_mirror_dst_port(config_db, interface_name):
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+    for k,v in mirror_table.items():
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+#
+# Check if port is already configured with mirror config
+#
+def interface_has_mirror_config(mirror_table, interface_name):
+    for k,v in mirror_table.items():
+        if 'src_port' in v and v['src_port'] == interface_name:
+            return True
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+
+#
+# Check if SPAN mirror-session config is valid.
+#
+def validate_mirror_session_config(config_db, session_name, dst_port, src_port, direction):
```

Comment:
Just checking, you can setup a mirror session with no ports specified simultaneously with a mirror session that is setup on specific interfaces, correct?

Reply:
Yes. we can specify with no source ports, and also with source ports. Sorry i am not sure i got the question correctly :)